operator/v1/types_insights.go

```diff
@@ -0,0 +1,124 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// enabled marks the health check as enabled
+	HealthCheckEnabled HealthCheckState = "Enabled"
+	// disabled marks the health check as disabled
+	HealthCheckDisabled HealthCheckState = "Disabled"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsOperator holds cluster-wide information about the Insights Operator.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type InsightsOperator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights.
+	// +kubebuilder:validation:Required
+	Spec InsightsOperatorSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator.
+	// +optional
+	Status InsightsOperatorStatus `json:"status"`
+}
+
+type InsightsOperatorSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsOperatorStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	// +optional
+	GatherStatus GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	// +optional
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
+	// +optional
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// lastGatherDuration is the last data gathering duration.
+	// +optional
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=atomic
+	// +optional
+	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
```

Comment:
I am thinking if we shouldn't call it `ActiveHealthChecks` instead.

Reply:
The `HealthCheck` struct includes a state which can be enabled/disabled, so perhaps some of the healthchecks listed will be inactive?

Reply:
OK yeah good point. I was thinking more in terms of a problem in a cluster = active recommendation, but yes it can be disabled by the user. Let's leave it as it is.